### PR TITLE
Enable derivation name duplicate with key

### PR DESCRIPTION
**Pull request by yuli-han**
## Summary

Currently building a derivation with name=key name is disabled by validator. But there are Airbnb users using this to introduce key to their response body. So introduce this PR to remove the validation on derivation name = key name. 

## Test Plan

- [x] Added Unit Tests
- [ ] Covered by existing CI
- [x] Integration tested

Testing by local tests on compiler

## Checklist
- [ ] Documentation update

## Reviewers
@hzding621 @pengyu-hou @donghanz

### CODE REVIEW

**Review by hzding621 — APPROVED** [review]

**Review by hzding621 — COMMENTED** [review]
@yuli-han Do you mind creating/updating unit tests?

File: api/py/test/sample/joins/sample_team/sample_join_external_parts.py [hunk 1]
```diff
@@ -52,12 +50,32 @@
...
-    table_properties={
-        "config_json": """{"sample_key": "sample_value"}"""
-    },
-    output_namespace="sample_namespace"
+    table_properties={"config_json": """{"sample_key": "sample_value"}"""},
+    derivations=[
+        Derivation(name="*", expression="*"),
+        Derivation(name="place_id", expression="ext_test_external_source_ext_place_id"),
+    ],
+    output_namespace="sample_namespace",
```
Comment (hzding621) [inline api/py/test/sample/joins/sample_team/sample_join_external_parts.py:80]:
```
v2 = Join(
    left=test_sources.staging_entities,
    right_parts=[JoinPart(group_by=sample_group_by.v1)],
    online_external_parts=[
        ExternalPart(
            ContextualSource(
                fields=[("place_id", DataType.STRING)]
            )
        )
    ],
    table_properties={"config_json": """{"sample_key": "sample_value"}"""},
    derivations=[
        Derivation(name="*", expression="*"),
        Derivation(name="place_id", expression="ext_contextual_place_id"),
    ],
    output_namespace="sample_namespace",
)
```

**Review by pkundurthy — APPROVED** [review]
LGTM

File: api/py/test/sample/joins/sample_team/sample_join_external_parts.py (hunk 1 above)
Comment (yuli-han) [reply api/py/test/sample/joins/sample_team/sample_join_external_parts.py:71]: updated

File: api/py/test/test_validator.py [hunk 2]
```diff
@@ -157,3 +157,11 @@ def test_validate_group_by_deprecation_date(zvalidator):
...
     errors = zvalidator._validate_group_by(v1_incorrect_deprecation_format)
     assert len(errors) == 1
+
+
+# Derive from external column to key should be enabled in validator
+def test_validate_derivation_on_keys(zvalidator):
+    from sample.joins.sample_team.sample_join_external_parts import v2
+
+    errors = zvalidator._validate_join(v2)
+    assert len(errors) == 0
```
[the remark this answers is not in the document]
Comment (yuli-han) [reply api/py/test/test_validator.py]: updated

**Pull request merged**